test/unit/test-amp-story-player.js

```diff
@@ -691,6 +691,85 @@ describes.realWin('AmpStoryPlayer', {amp: false}, (env) => {
       return expect(() => player.go(-1)).to.throw('Out of Story range.');
     });
 
+    it('last story call to next_() is the first story', async () => {
+      const playerEl = win.document.createElement('amp-story-player');
+      appendStoriesToPlayer(playerEl, 5);
+
+      const player = new AmpStoryPlayer(win, playerEl);
+
+      await player.load();
+      await nextTick();
+
+      playerEl.setAttribute('enable-circular-wrapping', '');
+      player.go(4);
+      swipeLeft();
+      const iframes = playerEl.shadowRoot.querySelectorAll('iframe');
+      await afterRenderPromise();
+      expect(iframes[0].getAttribute('i-amphtml-iframe-position')).to.eql('-1');
+      expect(iframes[1].getAttribute('i-amphtml-iframe-position')).to.eql('-1');
+      expect(iframes[2].getAttribute('i-amphtml-iframe-position')).to.eql('0');
```

Comment:
Now that we have the navigation event, we can use that to verify our tests instead of relying on the iframe positions. 

I played around with it and you can do something like this:


```suggestion
      expect(navigationSpy).to.have.been.calledWithMatch({
        type: 'navigation',
        detail: {
          index: 4,
          remaining: 0,
        },
      });
```

It would be great if you could also clean up the other tests that do this please!

Reply:
Done. There are still [three tests](https://github.com/ampproject/amphtml/blob/ce4569ee8d6561c59d625e1b8f84d542be84a8aa/test/unit/test-amp-story-player.js#L216-L263) that use iframe positions. Let me know if you want these changed along with this PR or another.